server/graphql/v2/mutation/EmojiReactionMutations.js

```diff
@@ -0,0 +1,164 @@
+    args: {
+      comment: {
+        type: CommentReferenceInput,
+      },
+      update: {
+        type: UpdateReferenceInput,
+      },
+      emoji: {
+        type: new GraphQLNonNull(GraphQLString),
+      },
+    },
+    resolve: async (_, { comment, update, emoji }, { remoteUser }) => {
+      mustBeLoggedInTo(remoteUser, 'remove this comment reaction');
+
+      if (!comment && !update) {
+        throw new Error('A comment or update must be provided');
+      }
+
+      if (comment) {
+        return removeReactionFromCommentOrUpdate(comment.id, remoteUser, emoji, IDENTIFIER_TYPES.COMMENT);
+      } else if (update) {
+        return removeReactionFromCommentOrUpdate(update.id, remoteUser, emoji, IDENTIFIER_TYPES.UPDATE);
+      }
+    },
+  },
+};
+
+const addReactionToCommentOrUpdate = async (id, req, emoji, identifierType) => {
+  const commentOrUpdateId = idDecode(id, identifierType);
+
+  let commentOrUpdate;
+  if (identifierType === IDENTIFIER_TYPES.COMMENT) {
+    commentOrUpdate = await models.Comment.findByPk(commentOrUpdateId);
+  } else {
+    commentOrUpdate = await models.Update.findByPk(commentOrUpdateId);
+  }
+
+  if (!commentOrUpdate) {
+    if (identifierType === IDENTIFIER_TYPES.COMMENT) {
+      throw new ValidationFailed('This comment does not exist');
+    } else {
+      throw new ValidationFailed('This update does not exist');
+    }
+  } else if (identifierType === IDENTIFIER_TYPES.COMMENT && commentOrUpdate.ExpenseId) {
+    const expense = await models.Expense.findByPk(commentOrUpdate.ExpenseId);
+    if (!expense || !(await canComment(req, expense))) {
+      throw new Forbidden('You are not allowed to comment or add reactions on this expense');
+    }
+  }
+
+  if (identifierType === IDENTIFIER_TYPES.COMMENT) {
+    const reaction = await models.EmojiReaction.addReactionOnComment(req.remoteUser, commentOrUpdateId, emoji);
+    return { comment: models.Comment.findByPk(reaction.CommentId), update: null };
+  } else {
+    const reaction = await models.EmojiReaction.addReactionOnUpdate(req.remoteUser, commentOrUpdateId, emoji);
+    return { update: models.Update.findByPk(reaction.UpdateId), comment: null };
+  }
+};
+
+const removeReactionFromCommentOrUpdate = async (id, remoteUser, emoji, identifierType) => {
+  const commentOrUpdateId = idDecode(id, identifierType);
+  let reaction;
+  if (identifierType === IDENTIFIER_TYPES.COMMENT) {
+    reaction = await models.EmojiReaction.findOne({
+      where: {
+        CommentId: commentOrUpdateId,
+        emoji,
+      },
+    });
+  } else {
+    reaction = await models.EmojiReaction.findOne({
+      where: {
+        UpdateId: commentOrUpdateId,
+        emoji,
+      },
+    });
+  }
+
+  if (!reaction) {
+    if (identifierType === IDENTIFIER_TYPES.COMMENT) {
+      throw new NotFound(`This comment reaction does not exist or has been deleted.`);
+    } else {
+      throw new NotFound(`This update reaction does not exist or has been deleted.`);
+    }
```

Comment:
:ice_cream: I think it's fine to have only one message here

```suggestion
    throw new NotFound(`This reaction does not exist or has been deleted.`);
```